airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -111,77 +117,102 @@ private void runInternal(final IntegrationConfig parsed) throws Exception {
-          final ConfiguredAirbyteCatalog catalog = parseConfig(parsed.getCatalogPath(), ConfiguredAirbyteCatalog.class);
-          try (final AirbyteMessageConsumer consumer = destination.getConsumer(config, catalog, outputRecordCollector)) {
-            runConsumer(consumer);
-          }
-        }
+        case WRITE -> runWrite(parsed);
         default -> throw new IllegalStateException("Unexpected value: " + parsed.getCommand());
       }
     } catch (final Exception e) {
-      // Many of the exceptions thrown are nested inside layers of RuntimeExceptions. An attempt is made
-      // to
-      // find the root exception that corresponds to a configuration error. If that does not exist, we
-      // just return the original exception.
-      final Throwable rootThrowable = ConnectorExceptionUtil.getRootConfigError(e);
-      final String displayMessage = ConnectorExceptionUtil.getDisplayMessage(rootThrowable);
-      // If the source connector throws a config error, a trace message with the relevant message should
-      // be surfaced.
-      if (ConnectorExceptionUtil.isConfigError(rootThrowable)) {
-        AirbyteTraceMessageUtility.emitConfigErrorTrace(e, displayMessage);
-      }
-      if (parsed.getCommand().equals(Command.CHECK)) {
-        // Currently, special handling is required for the CHECK case since the user display information in
-        // the trace message is
-        // not properly surfaced to the FE. In the future, we can remove this and just throw an exception.
-        outputRecordCollector
-            .accept(
-                new AirbyteMessage()
-                    .withType(Type.CONNECTION_STATUS)
-                    .withConnectionStatus(
-                        new AirbyteConnectionStatus()
-                            .withStatus(AirbyteConnectionStatus.Status.FAILED)
-                            .withMessage(displayMessage)));
-        return;
-      }
-      throw e;
+      handleRunInternalException(parsed, e);
     }
 
     LOGGER.info("Completed integration: {}", integration.getClass().getName());
   }
 
+  private AirbyteMessage runCheck(final IntegrationConfig parsed) throws Exception {
+    try {
+      final JsonNode config = getValidatedConfig(parsed, CHECK.toString());
+      return new AirbyteMessage().withType(Type.CONNECTION_STATUS).withConnectionStatus(integration.check(config));
+    } catch (final ConfigValidationException e) {
+      // if validation fails don't throw an exception, return a failed connection check message
+      return failedConnectionStatusMessage(e.getMessage());
+    }
+  }
+
+  private static boolean isCheckWithCatalog(IntegrationConfig parsed) {
+    return CHECK.equals(parsed.getCommand()) && parsed.getCatalogPath() != null;
+  }
+
+  private AirbyteMessage runCheckWithCatalog(final IntegrationConfig parsed) throws Exception {
+    LOGGER.info("Using Check With Catalog Method");
+    // TODO: Actually implement something different
```

Comment:
Is there an idea of what "something different" means?

Reply:
different that just calling the `runCheck` method, actually use the catalog for further validation.